crates/core_arch/src/x86_64/mod.rs

```diff
@@ -3,6 +3,26 @@
 #[macro_use]
 mod macros;
 
+#[unstable(feature = "x86_amx_intrinsics", issue = "126622")]
+#[allow(non_camel_case_types)]
+#[repr(packed)]
+#[derive(Copy, Clone, Default, Debug, PartialEq)]
+pub struct __tilecfg {
+    /// 0 `or` 1
+    pub palette: u8,
+    pub start_row: u8,
+    /// reserved, must be zero
+    reserved_a0: [u8; 14],
+    /// number of bytes of one row in each tile
+    pub colsb: [u16; 8],
+    /// reserved, must be zero
+    reserved_b0: [u16; 8],
+    /// number of rows in each tile
+    pub rows: [u8; 8],
+    /// reserved, must be zero
+    reserved_c0: [u8; 8],
+}
+
```

Comment:
Can you add a method in `__tilecfg` that takes the `pub` fields as input and returns the correctly configured cfg (as the `reserved` fields are private, this struct can't be constructed outside `core_arch`)?

Reply:
The GCC and Clang intrinsics don't define a `__tilecfg` type and just use `void *`. We shouldn't be inventing our own type here.

Reply:
Oh, thx for the notification, I did not notice the function signature of the ldtilecfg and sttilecfg in llvm which can be found [here](https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/IR/IntrinsicsX86.td). So I will make the `ldtilecfg` to accept `*const i8` instead of `*const __tilecfg` and similar to `sttilecfg`

Reply:
I think the __tilecfg is necessary to call amx intrinsics, so if we do not define it here, users might be forced to do so. How about we provide a usable definition here? Since the interface accepts `*const i8`, it is also okay for users to pass their self-defined struct to the function.